lib/connection.js

```diff
@@ -361,6 +361,14 @@ class Connection extends EventEmitter {
     this.emit('close', maybeErr);
   }
 
+  _updateSecret(newSecret, reason, cb) {
+    this.sendMethod(0, defs.ConnectionUpdateSecret, {
+      newSecret,
+      reason
+    });
+    this.on('update-secret-ok', cb);
```

Comment:
I'll change to this.once

Reply:
I used the event handler because I couldn't see another way of getting a callback into the channel0 accept function